components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -180,7 +189,16 @@ public ServiceProvider getApplicationExcludingFileBasedSPs(String applicationNam
             if (serviceProvider != null) {
                 loadApplicationPermissions(applicationName, serviceProvider);
             }
+
+            // invoking the listeners
+            for (ApplicationMgtListener listener : listeners) {
+                if (!listener.doPostGetApplication(serviceProvider, applicationName, tenantDomain)) {
+                    return null;
+                }
+            }
```

Comment:
Invoking listeners must be the final thing inside the service method. Just before the return. Also it must be outside any try catch blocks so that we don't need to unnecessarily handle any exception again in service method throw from listener.